python/Ganga/Core/GangaRepository/SessionLock.py

```diff
@@ -286,6 +270,28 @@ def decorated(self, *args, **kwargs):
             return f(self, *args, **kwargs)
     return decorated
 
+def global_disk_lock(f):
+    @functools.wraps(f)
+    def decorated_global(self, *args, **kwds):
+        with global_disk_lock.global_lock:
+            i=0
+            while i< 10:
+                i+=1
+                try:
+                    self.global_lock_acquire()
+                    break
+                except:
```

Comment:
I'll rewrite the global lock acquire to sanitize the exceptions it throws then. Only  way to know whats coming from there.